chain/chunks/src/lib.rs

```diff
@@ -368,25 +368,29 @@ impl ShardsManager {
         }
 
         for (account_id, part_ords) in bp_to_parts {
-            let request = PartialEncodedChunkRequestMsg {
-                chunk_hash: chunk_hash.clone(),
-                part_ords,
-                tracking_shards: if account_id == shard_representative_account_id {
-                    shards_to_fetch_receipts.clone()
-                } else {
-                    HashSet::new()
-                },
-            };
+            // extra check that we are not sending request to ourselves.
+            if Some(&account_id) != self.me.as_ref() {
```

Comment:
Can you add `else { debug_assert(false }`?

Having checks for conditions that are impossible without an assert is a bad practice IMO.